letsencrypt/acme/challenges_test.py

```diff
@@ -236,37 +234,37 @@ def test_json_without_optionals(self):
 class ProofOfPossessionHintsTest(unittest.TestCase):
 
     def setUp(self):
-        jwk = other.JWK(key=KEY.publickey())
-        issuers = [
+        jwk = jose.JWKRSA(key=KEY.publickey())
+        issuers = (
             'C=US, O=SuperT LLC, CN=SuperTrustworthy Public CA',
             'O=LessTrustworthy CA Inc, CN=LessTrustworthy But StillSecure',
-        ]
-        cert_fingerprints = [
+        )
```

Comment:
My aim was to make all protocol objects (messages, challenges, and anything else down the stack) hashable (`__hash__`), which enables to use those objects e.g. as keys in dictionaries (useful especially for server, which has to persist and match up session ACME messages/challenges). This in turn requires the objects to be immutable. That's why I'm using `tuple` over `list`, `frozendict` over `dict`, `frozenset` over `set`...